--- README.MPI
+++ README.MPI
@@ -0,0 +1,17 @@
+MPI-Enabled Utilities in Regina
+-------------------------------
+
+The following command-line utilities make use of MPI (the Message
+Passing Interface), which makes them well suited for use on clusters:
+
+- /usr/bin/tricensus-mpi  (form a census of 3-manifold triangulations)
+
+These utilities have been built against the LAM implementation of MPI.
+This means that they must be run under the LAM runtime environment
+(and not MPICH).  The LAM runtime environment can be found in the
+package "lam".
+
+If you have no need for MPI, Regina also ships with non-MPI alternatives
+to the above utilities.
+
+ -- Ben Burton <bab@debian.org>, Sun,  3 Jul 2005 18:35:25 +1000
