Index: general.h
===================================================================
--- general.h	(revision 7811)
+++ general.h	(working copy)
@@ -21,11 +21,15 @@
 
 
 #include <iostream>
+#include <cstring>
 #include <assert.h>
 
+/*
+// Regina will use GMP everywhere, even on Windows.
 #ifdef _WIN32 //for 32 and 64 bit windows
     #define NMZ_MPIR //always use MPIR
 #endif
+*/
 
 #ifdef NMZ_MPIR // use MPIR
     #include <mpirxx.h>
