--- general.h	(revision 7811)
+++ general.h	(working copy)
@@ -21,11 +21,15 @@
 
 
 #include <iostream>
+#include <cstring>
 #include <assert.h>
 
+/*
+// Regina will use GMP everywhere, even on Windows.
 #ifdef _WIN32 //for 32 and 64 bit windows
     #define NMZ_MPIR //always use MPIR
 #endif
+*/
 
 #ifdef NMZ_MPIR // use MPIR
     #include <mpirxx.h>
--- list_operations.h
+++ list_operations.h
@@ -41,7 +41,7 @@ ostream& operator<< (ostream& out, const list<T>& l) {
     for (i =l.begin(); i != l.end(); i++) {
         out << *i <<" ";
     }
-    out << endl;
+    out << std::endl;
     return out;
 }
 
