diff --git a/engine/libnormaliz/general.h b/engine/libnormaliz/general.h
index d17308e94..5370ec83c 100644
--- a/engine/libnormaliz/general.h
+++ b/engine/libnormaliz/general.h
@@ -54,9 +54,12 @@
 #include "libnormaliz/my_omp.h"
 
 
+/*
+// Regina will use GMP everywhere, even on Windows.
 #ifdef _WIN32     // for 32 and 64 bit windows
 #define NMZ_MPIR  // always use MPIR
 #endif
+*/
 
 #ifdef NMZ_MPIR  // use MPIR
 #include <mpirxx.h>
diff --git a/engine/libnormaliz/nmz_config.h b/engine/libnormaliz/nmz_config.h
index 2af3136b6..3b03f16e6 100644
--- a/engine/libnormaliz/nmz_config.h
+++ b/engine/libnormaliz/nmz_config.h
@@ -7,8 +7,8 @@
 #undef NMZ_FLINT
 #undef NMZ_COCOA
 
-#define ENFNORMALIZ
-#define NMZ_NAUTY
-#define NMZ_FLINT
-#define NMZ_COCOA
-#define NORMALIZ_USE_DLL
+// #define ENFNORMALIZ
+// #define NMZ_NAUTY
+// #define NMZ_FLINT
+// #define NMZ_COCOA
+// #define NORMALIZ_USE_DLL
