diff --git a/engine/libnormaliz/cone.cpp b/engine/libnormaliz/cone.cpp
index c09f3af9a..3145c2630 100644
--- a/engine/libnormaliz/cone.cpp
+++ b/engine/libnormaliz/cone.cpp
@@ -6270,8 +6270,10 @@ void Cone<Integer>::make_face_lattice(const ConeProperties& ToCompute){
             if(skip_remaining)
                 continue;
             
-           for(; kkk > Fpos; ++Fpos, ++F);
-           for(; kkk < Fpos; --Fpos, --F) ;
+           for(; kkk > Fpos; ++Fpos, ++F)
+               ;
+           for(; kkk < Fpos; --Fpos, --F)
+               ;
         
             if(verbose && nr_faces>=RepBound){
                 #pragma omp critical(VERBOSE)
diff --git a/engine/libnormaliz/general.h b/engine/libnormaliz/general.h
index c17ee6aa2..263a7ff6b 100644
--- a/engine/libnormaliz/general.h
+++ b/engine/libnormaliz/general.h
@@ -29,9 +29,12 @@
 #include <assert.h>
 #include <cstddef>
 
+/*
+// Regina will use GMP everywhere, even on Windows.
 #ifdef _WIN32 //for 32 and 64 bit windows
     #define NMZ_MPIR //always use MPIR
 #endif
+*/
 
 #ifdef NMZ_MPIR // use MPIR
     #include <mpirxx.h>
diff --git a/engine/libnormaliz/cone.cpp b/engine/libnormaliz/cone.cpp
index 46b8c8e05..120408b22 100644
--- a/engine/libnormaliz/cone.cpp
+++ b/engine/libnormaliz/cone.cpp
@@ -4478,6 +4478,7 @@ void Cone<Integer>::setFaceCodimBound(long bound){
     f_vector.clear();
 }
 
+#if 0
 bool executable(string command){
 //n check whether "command --version" cam be executed
 
@@ -4525,6 +4526,7 @@ string command(const string& original_call, const string& to_replace, const stri
         return test_path;
     return ""; // no executable found
 }
+#endif
 
 //---------------------------------------------------------------------------
 template<typename Integer>
diff --git a/engine/libnormaliz/descent.h b/engine/libnormaliz/descent.h
index 64940f119..61bc5ed83 100644
--- a/engine/libnormaliz/descent.h
+++ b/engine/libnormaliz/descent.h
@@ -29,6 +29,7 @@
 #include <list>
 #include <map>
 #include <boost/dynamic_bitset.hpp>
+#include "libnormaliz/matrix.h"
 
 namespace libnormaliz {
 using std::vector;
