Index: libtool.m4.ext
===================================================================
--- libtool.m4.ext	(revision 4731)
+++ libtool.m4.ext	(working copy)
@@ -37,7 +37,7 @@
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 ])
 
-# serial 56 LT_INIT
+# libtool.m4 serial 56 LT_INIT
 
 
 # LT_PREREQ(VERSION)
@@ -955,7 +955,7 @@
 	10.[[012]]*)
 	  _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
 	10.*)
-	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
+	  _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}dynamic_lookup' ;;
       esac
     ;;
   esac
Index: lt~obsolete.m4.ext
===================================================================
--- lt~obsolete.m4.ext	(revision 4731)
+++ lt~obsolete.m4.ext	(working copy)
@@ -7,7 +7,7 @@
 # unlimited permission to copy and/or distribute it, with or without
 # modifications, as long as this notice is preserved.
 
-# serial 4 lt~obsolete.m4
+# lt~obsolete.m4 serial 4 lt~obsolete.m4
 
 # These exist entirely to fool aclocal when bootstrapping libtool.
 #
Index: ltsugar.m4.ext
===================================================================
--- ltsugar.m4.ext	(revision 4731)
+++ ltsugar.m4.ext	(working copy)
@@ -7,7 +7,7 @@
 # unlimited permission to copy and/or distribute it, with or without
 # modifications, as long as this notice is preserved.
 
-# serial 6 ltsugar.m4
+# ltsugar.m4 serial 6 ltsugar.m4
 
 # This is to help aclocal find these macros, as it can't see m4_define.
 AC_DEFUN([LTSUGAR_VERSION], [m4_if([0.1])])
Index: ltversion.m4.ext
===================================================================
--- ltversion.m4.ext	(revision 4731)
+++ ltversion.m4.ext	(working copy)
@@ -9,7 +9,7 @@
 
 # Generated from ltversion.in.
 
-# serial 3012 ltversion.m4
+# ltversion.m4 serial 3012 ltversion.m4
 # This file is part of GNU Libtool
 
 m4_define([LT_PACKAGE_VERSION], [2.2.6])
Index: ltoptions.m4.ext
===================================================================
--- ltoptions.m4.ext	(revision 4731)
+++ ltoptions.m4.ext	(working copy)
@@ -7,7 +7,7 @@
 # unlimited permission to copy and/or distribute it, with or without
 # modifications, as long as this notice is preserved.
 
-# serial 6 ltoptions.m4
+# ltoptions.m4 serial 6 ltoptions.m4
 
 # This is to help aclocal find these macros, as it can't see m4_define.
 AC_DEFUN([LTOPTIONS_VERSION], [m4_if([1])])
Index: kde.m4.ext
===================================================================
--- kde.m4.ext	(revision 3877)
+++ kde.m4.ext	(working copy)
@@ -3308,7 +3308,7 @@
 # Search path for a program which passes the given test.
 # Ulrich Drepper <drepper@cygnus.com>, 1996.
 
-# serial 1
+# kde.m4 serial 1
 # Stephan Kulow: I appended a _KDE against name conflicts
 
 dnl AM_PATH_PROG_WITH_TEST_KDE(VARIABLE, PROG-TO-CHECK-FOR,
@@ -3353,7 +3353,7 @@
 # Check whether LC_MESSAGES is available in <locale.h>.
 # Ulrich Drepper <drepper@cygnus.com>, 1995.
 
-# serial 1
+# kde.m4 serial 1
 
 AC_DEFUN([AM_LC_MESSAGES],
   [if test $ac_cv_header_locale_h = yes; then
@@ -3398,7 +3398,7 @@
 # Macro to add for using GNU gettext.
 # Ulrich Drepper <drepper@cygnus.com>, 1995.
 
-# serial 1
+# kde.m4 serial 1
 # Stephan Kulow: I put a KDE in it to avoid name conflicts
 
 AC_DEFUN([AM_KDE_GNU_GETTEXT],
