diff --git a/engine/snappea/kernel/SnapPea.h b/engine/snappea/kernel/SnapPea.h
index bb7ccad80..6fa6aaa96 100644
--- a/engine/snappea/kernel/SnapPea.h
+++ b/engine/snappea/kernel/SnapPea.h
@@ -496,9 +496,11 @@ typedef struct NormalSurfaceList            NormalSurfaceList;
 /* #define CONST const */
 
 
+/* We build the snappea kernel as C++, so link it as C++. - B.B., 27/03/2014.
 #ifdef __cplusplus
 extern "C" {
 #endif
+*/
 
 /************************************************************************/
 
@@ -2697,9 +2699,11 @@ extern Real volume(Triangulation *manifold, int *precision);
 
 #include "end_namespace.h"
 
+/* We build the snappea kernel as C++, so link it as C++. - B.B., 27/03/2014.
 #ifdef __cplusplus
 }
 #endif
+*/
 
 #endif
 /* Local Variables:                      */
diff --git a/engine/snappea/kernel/end_namespace.h b/engine/snappea/kernel/end_namespace.h
index 7f7c29f1b..c0911805e 100644
--- a/engine/snappea/kernel/end_namespace.h
+++ b/engine/snappea/kernel/end_namespace.h
@@ -5,3 +5,5 @@
  *  If you declared a namespace block in kernel_namespace.h then
  *  you can close the block here.
  */
+
+} }
diff --git a/engine/snappea/kernel/kernel.h b/engine/snappea/kernel/kernel.h
index 7b97fd785..837f6d1c7 100644
--- a/engine/snappea/kernel/kernel.h
+++ b/engine/snappea/kernel/kernel.h
@@ -11,9 +11,11 @@
 
 #include "SnapPea.h"
 
+/* We build the snappea kernel as C++, so link it as C++. - B.B., 27/03/2014.
 #ifdef __cplusplus
 extern "C" {
 #endif
+*/
 
 #include <string.h>
 #include <math.h>
@@ -30,9 +32,11 @@ extern "C" {
 #include "kernel_prototypes.h"
 #include "tables.h"
 
+/* We build the snappea kernel as C++, so link it as C++. - B.B., 27/03/2014.
 #ifdef __cplusplus
 }
 #endif
+*/
 
 #endif
 /* Local Variables:                      */
diff --git a/engine/snappea/kernel/kernel_namespace.h b/engine/snappea/kernel/kernel_namespace.h
index c24d6199e..ef142f978 100644
--- a/engine/snappea/kernel/kernel_namespace.h
+++ b/engine/snappea/kernel/kernel_namespace.h
@@ -16,3 +16,5 @@
 #endif
 #endif
 #endif
+
+namespace regina { namespace snappea {
