diff --git a/engine/snappea/kernel/homology.h b/engine/snappea/kernel/homology.h
index c4ae28b..8548561 100644
--- a/engine/snappea/kernel/homology.h
+++ b/engine/snappea/kernel/homology.h
@@ -38,9 +38,9 @@
  */
 
 typedef long int MatrixEntry;
-#define ENTRY_MAX   LLONG_MAX
-#define ENTRY_MIN   (-LLONG_MAX)
-#define LIMIT_MIN   LLONG_MIN
+#define ENTRY_MAX   LONG_MAX
+#define ENTRY_MIN   (-LONG_MAX)
+#define LIMIT_MIN   LONG_MIN
 
 /*
  *  The number of meaningful rows and columns in a RelationMatrix are
diff --git a/engine/snappea/kernel/o31_matrices.c b/engine/snappea/kernel/o31_matrices.c
index 23f45cb..8945d47 100644
--- a/engine/snappea/kernel/o31_matrices.c
+++ b/engine/snappea/kernel/o31_matrices.c
@@ -325,10 +325,10 @@ void o31_product(
     O31Matrix   b,
     O31Matrix   product)
 {
-    register int    i,
-                    j,
-                    k;
-    register Real sum;
+    int    i,
+           j,
+           k;
+    Real sum;
     O31Matrix       temp;
 
     for (i = 0; i < 4; i++)
@@ -522,9 +522,9 @@ void o31_matrix_times_vector(
     O31Vector   v,
     O31Vector   product)
 {
-    register int    i,
-                    j;
-    register Real sum;
+    int    i,
+           j;
+    Real sum;
     O31Vector       temp;
 
     for (i = 0; i < 4; i++)
diff --git a/engine/snappea/kernel/solve_equations.c b/engine/snappea/kernel/solve_equations.c
index 77df264..43bf717 100644
--- a/engine/snappea/kernel/solve_equations.c
+++ b/engine/snappea/kernel/solve_equations.c
@@ -50,15 +50,15 @@ FuncResult solve_complex_equations(
     Complex *solution)
 {
     /*
-     *  The following register variables are used in the n^3 bottleneck.
+     *  The following variables are used in the n^3 bottleneck.
      *  (See below.)
      */
 
-    register Real       factor_real,
-                        factor_imag;
-    register Complex    *row_r,
-                        *row_c;
-    register int        count;
+    Real       factor_real,
+               factor_imag;
+    Complex    *row_r,
+               *row_c;
+    int        count;
 
     /*
      *  The remaining variables are used in less critical places.
@@ -252,14 +252,14 @@ FuncResult solve_real_equations(
     Real  *solution)
 {
     /*
-     *  The following register variables are used in the n^3 bottleneck.
+     *  The following variables are used in the n^3 bottleneck.
      *  (See below.)
      */
 
-    register Real factor,
-                    *row_r,
-                    *row_c;
-    register int    count;
+    Real factor,
+           *row_r,
+           *row_c;
+    int    count;
 
     /*
      *  The remaining variables are used in less critical places.
